### ISSUE TITLE
Can't seem to connect Github repo to Railway??

### ISSUE THREAD

**Issue by jameszenartist** [railwayapp/cli#289]
So I've been working on a PERN stack application, but I'm having trouble connecting my existing repo to Railway.

Every attempt ends with this window and doesn't move forward?

I checked my settings in my Github account, and all is well there. I was curious if anyone else has been experiencing the same?

I would presume at least some of the developers at Railway already know about this behavior? If so, I was wondering if anyone might be able to provide some insight?

I also found this on SO, and it looks like someone else is dealing with the same?:
[https://stackoverflow.com/questions/73965356/railway-configuring-repo-access/73965453#73965453](here)

This appears to be a wonderful service, and I very much am looking forward to using it more often!

Thanks for reading! 
 -James
![railway-github](https://user-images.githubusercontent.com/41663318/202855292-e32e4420-4c9b-4515-9825-07640a317976.png)

**Comment by Tijani-Dia** [comment]
I can confirm that I've experienced the exact same issue today.

**Comment by f-a-t-h-e** [comment]
Same issue and I'm trying to use CircleCI because even github actions don't work.

**Comment by f-a-t-h-e** [comment]
I've reached a manual solution.

### using Github actions

-Edit action.yaml file to look like this and don't forget to change your-project-id to your railway project id
`name: Deploy to Railway

on:
  push:
    branches: [main]

jobs:
  deploy:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Use Node 16
        uses: actions/setup-node@v1
        with:
          node-version: 16.x

      - name: Install packages
        run: yarn install --frozen-lockfile

      - name: Install Railway
        run: npm i -g @railway/cli

      - name: Login to Railway
        run: railway login --browserless

      - name: Link the project to railway
        run: railway link your-project-id

      - name: Deploy
        run: railway up
        env:
          RAILWAY_TOKEN: ${{ env.RAILWAY_TOKEN }}

`

-Set the environment variable in your github repo `RAILWAY_TOKEN` to the value that you can generate in your railway project


### Using CircleCI

-Edit config.yaml file to look like this and don't forget to change your-project-id to your railway project id

`
version: 2.1
name: Node.js CI
orbs:
  node: circleci/node@5.0.2
jobs:
  build:
    docker:
      - image: "cimg/node:16.10"
    steps:
      - node/install:
          node-version: "16.10"
      - checkout
      - run: echo $(node --version)
      - run: echo $(npm --version)
      - run: echo "The present working directory is $( pwd; )"
      - run: yarn
      - run: npm run lint
      - run: npm run build
      - run: npm test
  deploy:
    docker:
      - image: "cimg/node:16.10"
    steps:
      # install node
      - node/install:
          node-version: "16.10"
      - checkout
      - run: yarn install --frozen-lockfile
      - run: yarn test
      - run: npm i -g @railway/cli
      - run: railway login --browserless
      - run: railway link your-project-id
      - run: railway up
workflows:
  nest-prisma:
    jobs:
      - build:
          filters:
            branches:
              only:
                - main
      - hold:
          filters:
            branches:
              only:
                - main
          type: approval
          requires:
            - build
      - deploy:
          filters:
            branches:
              only:
                - main
          requires:
            - hold
`

-Set the environment variable in your CircleCi project `RAILWAY_TOKEN` to the value that you can generate in your railway project.


I hope this helps and if anybody has a better solution (except deploying from the local machine) please share it

**Comment by f-a-t-h-e** [comment]
> 

##references: 
-

**Comment by f-a-t-h-e** [comment]
> I've reached a manual solution.

### References:

- [Railway docs](https://docs.railway.app/deploy/integrations#project-tokens)
- [This tutorial by Muhammed Ali](https://adamtheautomator.com/github-actions-environment-variables/)
- [This Post by Damien Aicheh](https://damienaicheh.github.io/github/actions/2021/04/15/environment-variables-secrets-github-actions-en.html)
- [This blog by Faraz Patankar](https://blog.railway.app/p/github-actions)

- [this file from Udacity starter project](https://github.com/udacity/nd0067-c4-deployment-process-project-starter/blob/master/.circleci/config.yml)

**Comment by jameszenartist** [comment]
Greetings @f-a-t-h-e ,

Thank you so much for the quick responses and feedback!

I'm going to go over your reference material and give this a try as soon as I can.

I'll let you and everyone else who's invloved know of my results.

Thanks again!,

-James

**Comment by jameszenartist** [comment]
Hi @f-a-t-h-e,

I believe I finally solved the issue!

It dawned on me as I was tinkering with connecting to other platforms like Vercel and Render, and was noticing some similar behaviour. 

While your suggestion unfortunately didn't solve the problem, I found that if you delete the me50 connection (if there is one of course) on your Github originizations page, the loading error goes away!

I hope this helps everyone as well!

Thanks again for your advice :) 

 -James
